app/lib/features/tasks/widgets/task_list_item_card.dart

```diff
@@ -1,50 +1,64 @@
 import 'package:acter/common/providers/room_providers.dart';
 import 'package:acter/common/utils/routes.dart';
+import 'package:acter/features/tasks/providers/tasklists_providers.dart';
 import 'package:acter/features/tasks/widgets/task_items_list_widget.dart';
 import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:go_router/go_router.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::tasks::widget::task_list_item');
 
 class TaskListItemCard extends ConsumerWidget {
-  final TaskList taskList;
+  final String taskListId;
   final bool showSpace;
   final bool showCompletedTask;
   final bool initiallyExpanded;
 
   const TaskListItemCard({
     super.key,
-    required this.taskList,
+    required this.taskListId,
     this.showSpace = false,
     this.showCompletedTask = false,
     this.initiallyExpanded = true,
   });
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final taskListId = taskList.eventIdStr();
-    return Card(
-      key: Key('task-list-card-$taskListId'),
-      child: ExpansionTile(
-        initiallyExpanded: initiallyExpanded,
-        shape: RoundedRectangleBorder(
-          borderRadius: BorderRadius.circular(10),
-        ),
-        iconColor: Theme.of(context).colorScheme.onSurface,
-        childrenPadding: const EdgeInsets.symmetric(horizontal: 10),
-        title: title(context),
-        subtitle: subtitle(ref),
-        children: [
-          TaskItemsListWidget(
-            taskList: taskList,
-            showCompletedTask: showCompletedTask,
+    return ref.watch(taskListItemProvider(taskListId)).when(
+          data: (taskList) => Card(
+            key: Key('task-list-card-$taskListId'),
+            child: ExpansionTile(
+              initiallyExpanded: initiallyExpanded,
+              shape: RoundedRectangleBorder(
+                borderRadius: BorderRadius.circular(10),
+              ),
+              iconColor: Theme.of(context).colorScheme.onSurface,
+              childrenPadding: const EdgeInsets.symmetric(horizontal: 10),
+              title: title(context, taskList),
+              subtitle: subtitle(ref, taskList),
+              children: [
+                TaskItemsListWidget(
+                  taskList: taskList,
+                  showCompletedTask: showCompletedTask,
+                ),
+              ],
+            ),
           ),
-        ],
-      ),
-    );
+          error: (error, stack) {
+            _log.severe('failed to load tasklist', error, stack);
+            return const Card(
+              child: Text('failed'),
+            );
+          },
+          loading: () => const Card(
+            child: Text('loading'),
+          ),
+        );
   }
 
-  Widget title(BuildContext context) {
+  Widget title(BuildContext context, TaskList taskList) {
     final taskListId = taskList.eventIdStr();
```

Comment:
unnecessary, we already know `taskListId`.
```suggestion
  Widget title(BuildContext context, TaskList taskList) {
```